src/java.base/share/classes/java/util/Random.java

```diff
@@ -466,10 +466,19 @@ protected int next(int bits) {
      */
     @Override
     public void nextBytes(byte[] bytes) {
-        for (int i = 0, len = bytes.length; i < len; )
-            for (int rnd = nextInt(),
-                 n = Math.min(len - i, Integer.SIZE/Byte.SIZE);
-                 n-- > 0; rnd >>= Byte.SIZE)
+        int i = 0;
+        int len = bytes.length;
+
+        for (int words = len >> 2; words--> 0; ) {
+            int rnd = nextInt();
+            if (unsafe.isBigEndian())
+                rnd = Integer.reverseBytes(rnd);
+            unsafe.putInt(bytes, (long)Unsafe.ARRAY_BYTE_BASE_OFFSET + i, rnd);
```

Comment:
Isn't there a `putInt` that accepts another trailing boolean argument, indicating if the value is big endian?
```java
        int len = bytes.length;

        for (int words = len / Integer.BYTES; words > 0; words--) {
            unsafe.putInt(bytes, Unsafe.ARRAY_BYTE_BASE_OFFSET + i, nextInt(), false);
        }

        int i = len & (~0x3); // len / Integer.BYTES * Integer.BYTES
        if (i < len)
            for (int rnd = nextInt(); i < len; rnd >>>= Byte.SIZE)
                bytes[i++] = (byte)rnd;
```
I would recommend updating implementation for RandomGenerator to remove its manual increment of i.